app/lib/common/providers/notifiers/sync_notifier.dart

```diff
@@ -3,33 +3,47 @@ import 'dart:math';
 
 import 'package:acter/common/extensions/options.dart';
 import 'package:acter/common/models/sync_state/sync_state.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
 import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart' as ffi;
 import 'package:riverpod/riverpod.dart';
 import 'package:sentry_flutter/sentry_flutter.dart';
 
 // ignore_for_file: avoid_print
-class SyncNotifier extends StateNotifier<SyncState> {
-  final ffi.Client client;
-  final Ref ref;
+class SyncNotifier extends Notifier<SyncState> {
+  late ffi.Client client;
 
   late ffi.SyncState syncState;
-  late Stream<bool> _syncListener;
-  late StreamSubscription<bool> _syncPoller;
-  late Stream<String> _errorListener;
-  late StreamSubscription<String> _errorPoller;
+  Stream<bool>? _syncListener;
+  StreamSubscription<bool>? _syncPoller;
+  Stream<String>? _errorListener;
+  StreamSubscription<String>? _errorPoller;
+  late ProviderSubscription _providerSubscription;
   Timer? _retryTimer;
 
-  SyncNotifier(this.client, this.ref)
-      : super(const SyncState(initialSync: true)) {
-    _startSync(ref);
-  }
-
-  void _startSync(Ref ref) {
-    // on release we have a really weird behavior, where, if we schedule
-    // any async call in rust too early, they just pend forever. this
-    // hack unfortunately means we have two wait a bit but that means
-    // we get past the threshold where it is okay to schedule...
-    Future.delayed(const Duration(milliseconds: 1500), () => _restartSync());
+  @override
+  SyncState build() {
+    _providerSubscription = ref.listen<AsyncValue<ffi.Client?>>(
+      alwaysClientProvider,
+      (AsyncValue<ffi.Client?>? oldVal, AsyncValue<ffi.Client?> newVal) {
+        final newClient = newVal.valueOrNull;
+        if (newClient == null) {
+          // we don't care for not having a proper client yet
+          return;
+        }
+        // on release we have a really weird behavior, where, if we schedule
+        // any async call in rust too early, they just pend forever. this
+        // hack unfortunately means we have two wait a bit but that means
+        // we get past the threshold where it is okay to schedule...
+        client = newClient;
+        Future.delayed(
+          const Duration(milliseconds: 1500),
+          () => _restartSync(),
+        );
```

Comment:
it stands to be seen whether this comment still applies now and thus the delay is necessary or whether that means we can now remove that as all this happens "later" anyways.